tests/sql/restore_incidents_test.go

```diff
@@ -0,0 +1,254 @@
+package sql_test
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"github.com/creasty/defaults"
+	"github.com/icinga/icinga-notifications/internal/config"
+	"github.com/icinga/icinga-notifications/internal/event"
+	"github.com/icinga/icinga-notifications/internal/incident"
+	"github.com/icinga/icinga-notifications/internal/object"
+	baseConfig "github.com/icinga/icingadb/pkg/config"
+	"github.com/icinga/icingadb/pkg/icingadb"
+	"github.com/icinga/icingadb/pkg/logging"
+	"github.com/icinga/icingadb/pkg/types"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestIncidentLoading(t *testing.T) {
+	t.Parallel()
```

Comment:
You should be careful with `t.Parallel()` here. If the test uses a shared database, you might not want other tests to run at the same time.